### DISCUSSION TITLE
Bitbucket server + project/repository HTTP access token auth supported?

### DISCUSSION THREAD

**Discussion by xpacm** [renovatebot/renovate#19211] [Request Help]
Hi everybody,

I'm getting 401s when trying to authenticate to a Bitbucket server using an HTTP access token created for a particular repository. Bitbucket docs recommend using only Bearer Auth without username in such cases. Indeed, curl tests are successful when using this method, and also give 401 when using Basic Auth.

Is such auth mode currently supported by renovate, and if yes, which config auth settings shoud I use?

Thanks,
Alex

**Comment by viceice** [comment]
i think you need to configure the token as password

**Reply by xpacm** [reply]
```
module.exports = {
  platform: 'bitbucket-server',
  endpoint: 'https://my.bitbucket.server',
  username: 'my_username',
  password: 'http_token_for_the_repo',
  dryRun: 'full',
  repositories: ['MA/android-client-renovate-test'],
};
```

You mean like this? This gives 401. Tried with an http access token created for the account - it works.

**Comment by rarkins** [comment]
I noticed that we don't mention "token" in the lib/modules/platforms/bitbucket/readme.md file, however I recall something similar was added in https://github.com/renovatebot/renovate/pull/16900

**Reply by rarkins** [reply]
Sorry that was for Bitbucket Cloud

**Comment by xpacm** [comment]
Bitbucket version: Atlassian Bitbucket v7.21.4

I'm pretty sure Basic Auth is not supported for project/repo HTTP access tokens by at least this particular Bitbucket version.

`curl -v -u my_username:repo_http_token https://my.bitbucket.server/rest/api/1.0/projects/MA/repos/android-client-renovate-test` - gives 401.

`curl -v -H 'Authorization: Bearer repo_http_token' https://my.bitbucket.server/rest/api/1.0/projects/MA/repos/android-client-renovate-test` - works.

**Reply by viceice** [reply]
Then it needs adaptions in `bitbucket-server` platform code to allow `token` auth additionally to `username` and `password`
